plugins/woocommerce/src/Database/Migrations/CustomOrderTable/CLIRunner.php

```diff
@@ -0,0 +1,412 @@
+
+use Automattic\WooCommerce\Internal\DataStores\Orders\CustomOrdersTableController;
+use Automattic\WooCommerce\Internal\DataStores\Orders\DataSynchronizer;
+use WP_CLI;
+
+/**
+ * CLI tool for migrating order data to/from custom table.
+ *
+ * Credits https://github.com/liquidweb/woocommerce-custom-orders-table/blob/develop/includes/class-woocommerce-custom-orders-table-cli.php.
+ *
+ * Class CLIRunner
+ */
+class CLIRunner {
+
+	/**
+	 * CustomOrdersTableController instance.
+	 *
+	 * @var CustomOrdersTableController
+	 */
+	private $controller;
+
+	/**
+	 * DataSynchronizer instance.
+	 *
+	 * @var DataSynchronizer;
+	 */
+	private $synchronizer;
+
+	/**
+	 * PostsToOrdersMigrationController instance.
+	 *
+	 * @var PostsToOrdersMigrationController
+	 */
+	private $post_to_cot_migrator;
+
+	/**
+	 * Init method, invoked by DI container.
+	 *
+	 * @param CustomOrdersTableController      $controller Instance.
+	 * @param DataSynchronizer                 $synchronizer Instance.
+	 * @param PostsToOrdersMigrationController $posts_to_orders_migration_controller Instance.
+	 *
+	 * @internal
+	 */
+	final public function init( CustomOrdersTableController $controller, DataSynchronizer $synchronizer, PostsToOrdersMigrationController $posts_to_orders_migration_controller ) {
+		$this->controller           = $controller;
+		$this->synchronizer         = $synchronizer;
+		$this->post_to_cot_migrator = $posts_to_orders_migration_controller;
+	}
+
+	/**
+	 * Registers commands for CLI.
+	 */
+	public function register_commands() {
+		WP_CLI::add_command( 'wc cot count', array( $this, 'count' ) );
+		WP_CLI::add_command( 'wc cot migrate', array( $this, 'migrate' ) );
+		WP_CLI::add_command( 'wc cot verify_cot_data', array( $this, 'verify_cot_data' ) );
+	}
+
+	/**
+	 * Check if the COT feature is enabled.
+	 *
+	 * @param bool $log Optionally log a error message.
+	 *
+	 * @return bool Whether the COT feature is enabled.
+	 */
+	private function is_enabled( $log = true ) {
+		if ( ! $this->controller->is_feature_visible() ) {
+			if ( $log ) {
+				WP_CLI::log(
+					sprintf(
+						// translators: %s - link to testing instructions webpage.
+						__( 'Custom order table usage is not enabled. If you are testing, you can enable it by following the testing instructions in %s', 'woocommerce' ),
+						'https://developer.woocommerce.com/' // TODO: Change the link when testing instructin page is live.
+					)
+				);
+			}
+		}
+
+		return $this->controller->is_feature_visible();
+	}
+
+	/**
+	 * Helper method to log warning that feature is not yet production ready.
+	 */
+	private function log_production_warning() {
+		WP_CLI::log( __( 'This feature is not production ready yet. Make sure you are not running these commands in your production environment.', 'woocommerce' ) );
+	}
+
+	/**
+	 * Count how many orders have yet to be migrated into the custom orders table.
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     wp wc cot count
+	 *
```

Comment:
```suggestion
	 *
	 * ## OPTIONS
	 *
	 * [--log=<log>]
	 * : If a message describing the number of orders to be migrated should be output.
	 * ---
	 * default: 1
	 * options:
	 *   - 0
	 *   - 1
	 *
```

Should we document how `--log` works (something like the above, specifying 1 and 0 since `--log=false` actually evaluates as log being true) ... or prefer to keep as a private implementation detail?

Reply:
This is actually an internal method, as I don't think anyone would invoke this directly but then won't expect for log to be printed.

Reply:
Gotcha gotcha